base/c.jl

```diff
@@ -737,3 +737,73 @@ end
 macro ccall_effects(effects::UInt8, expr)
     return ccall_macro_lower((:ccall, effects), ccall_macro_parse(expr)...)
 end
+
+
+"""
+    @cglobal [library.]symbol[::type=Cvoid]
+    @cglobal \$sym_ptr[::type=Cvoid]
+
+Obtain a pointer to a global variable in a C-exported shared library, specified as in [`@ccall`](@ref]).
+
+See also: [`cglobal`](@ref).
+
+# Examples
+
+```julia-repl
+julia> @cglobal errno::Cint
+Ptr{Int32} @0x00007ffaf3c396b8
+
+julia> @cglobal "./mylib.so".y::Cint
+Ptr{Int32} @0x00007ffaf3246028
+
+julia> libh = Libdl.dlopen("libccalltest");
+
+julia> sym = Libdl.dlsym(libh, :global_var)
+Ptr{Nothing} @0x00007ffaf3239190
+
+julia> @cglobal \$sym::Cint
+Ptr{Int32} @0x00007f08380c9190
```

Comment:
What is this doing? Isn't `cglobal` just a bitcast if the argument is already a pointer?

Reply:
Yes, but `cglobal` seems to support it so I added this too.

Reply:
More specifically, these tests https://github.com/JuliaLang/julia/blob/902a5c199d590552758f7a91cc75c47ea67de5f2/test/ccall.jl#L1857-L1866 test it.